collect_app/src/main/java/org/odk/collect/android/widgets/GeoPointWidget.java

```diff
@@ -18,152 +18,183 @@
         if (stringAnswer == null || stringAnswer.isEmpty()) {
             return null;
         } else {
-            try {
-                return new GeoPointData(GeoWidgetUtils.getLocationParamsFromStringAnswer(stringAnswer));
-            } catch (Exception numberFormatException) {
-                return null;
-            }
+            return new GeoPointData(GeoWidgetUtils.getLocationParamsFromStringAnswer(stringAnswer));
         }
     }
 
     @Override
     public void clearAnswer() {
         stringAnswer = null;
-        super.clearAnswer();
+        binding.geoAnswerText.setText(null);
+        updateButtonLabelsAndVisibility(false);
+        widgetValueChanged();
+    }
+
+    @Override
+    public void setOnLongClickListener(OnLongClickListener l) {
+        binding.simpleButton.setOnLongClickListener(l);
+        binding.geoAnswerText.setOnLongClickListener(l);
+    }
+
+    @Override
+    public void cancelLongPress() {
+        super.cancelLongPress();
+        binding.simpleButton.cancelLongPress();
+        binding.geoAnswerText.cancelLongPress();
     }
 
     @Override
     public void setBinaryData(Object answer) {
         stringAnswer = (String) answer;
+        binding.geoAnswerText.setText(getAnswerToDisplay(stringAnswer));
 
-        if (stringAnswer != null && !stringAnswer.isEmpty()) {
-            String[] parts = stringAnswer.split(" ");
-            answerDisplay.setText(getContext().getString(
-                R.string.gps_result,
-                GeoWidgetUtils.convertCoordinatesIntoDegreeFormat(getContext(), Double.parseDouble(parts[0]), "lat"),
-                GeoWidgetUtils.convertCoordinatesIntoDegreeFormat(getContext(), Double.parseDouble(parts[1]), "lon"),
-                    GeoWidgetUtils.truncateDouble(parts[2]),
-                    GeoWidgetUtils.truncateDouble(parts[3])
-            ));
-        } else {
-            answerDisplay.setText("");
+        if (binding.geoAnswerText.getText().toString().equals("")) {
+            stringAnswer = "";
         }
 
-        updateButtonLabelsAndVisibility(true);
+        updateButtonLabelsAndVisibility(stringAnswer != null);
         widgetValueChanged();
     }
 
-    @Override
-    public String getAnswerToDisplay(String answer) {
-        if (!answer.isEmpty()) {
-            String[] parts = answer.split(" ");
-            return getContext().getString(
-                    R.string.gps_result,
-                    GeoWidgetUtils.convertCoordinatesIntoDegreeFormat(getContext(), Double.parseDouble(parts[0]), "lat"),
-                    GeoWidgetUtils.convertCoordinatesIntoDegreeFormat(getContext(), Double.parseDouble(parts[1]), "lon"),
-                    GeoWidgetUtils.truncateDouble(parts[2]),
-                    GeoWidgetUtils.truncateDouble(parts[3])
-            );
-        } else {
+    protected GeoWidgetAnswerBinding getBinding() {
+        return binding;
+    }
+
+    private void determineMapProperties(QuestionDef questionDef) {
```

Comment:
There's no map so the name isn't right. I think that here the code can either go directly in the constructor or be renamed to something like `getAccuracyThreshold`. If it's the later, then I think the method should return a `double`.